android/PlayKit/app/src/main/java/io/streamroot/dna/playkit/PlayKitInteractor.java

```diff
@@ -0,0 +1,54 @@
+package io.streamroot.dna.playkit;
+
+import android.os.Looper;
+import com.kaltura.playkit.Player;
+import io.streamroot.dna.core.PlayerInteractor;
+import io.streamroot.dna.core.TimeRange;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collections;
+import java.util.List;
+
+public final class PlayKitInteractor implements PlayerInteractor {
+    private Player mPlayer;
+
+    public PlayKitInteractor(Player player) {
+        this.mPlayer = player;
+    }
+
+    @Override
+    public double bufferTarget() {
+        Double bufferTarget = mPlayer.getBufferTarget();
+
+        if (bufferTarget != null) {
+            return bufferTarget;
+        }
+
+        return 0.0;
+    }
+
+    @NotNull
+    @Override
+    public List<TimeRange> loadedTimeRanges() {
+        long currentPosition = mPlayer.getCurrentPosition();
+        long bufferedPosition = mPlayer.getBufferedPosition();
+        return Collections.singletonList(new TimeRange(currentPosition, bufferedPosition));
+    }
+
+    @Nullable
+    @Override
+    public Looper looper() {
+        return Looper.getMainLooper();
+    }
+
+    @Override
+    public long playbackTime() {
+        return mPlayer.getCurrentPosition();
+    }
+
+    @Override
+    public void setBufferTarget(double target) {
+        mPlayer.setBufferTarget(target);
```

Comment:
did they implement it for us? is it in second?

Reply:
It's the API added by our `dna-playkit-android` fork 😉 (and AFAIR I used same call convention in the `ExoPlayer` interactor)

Reply:
ah alright 👍